### PR TITLE
Multiple fixes to GroupedGemm+SplitK

### PR DESCRIPTION
**Pull request by aosewski**
This PR:

* Add use of GridwiseGemmPipeline to `GridwiseGemm_bk0mk1_bk0nk1_mn_xdlops_v2r4r2` kernel to select proper version and have correct validation.
* Add multiple debug log messages.
* Fix issue with MN padding when creating CGridDescriptor
* Add multiple unit-tests for different KBatch values.
* Change gemm-splitk unit test to leverage ckProfiler API
* Add Interface tests for grouped gemm

### CODE REVIEW

File: test/gemm_split_k/test_gemm_splitk_ut_cases.inc [hunk 1]
```diff
@@ -0,0 +1,217 @@
+#pragma once
+
+TYPED_TEST(TestGemmSplitK_MK_KN, SmallM)
+{
+    std::vector<int> Ms{0, 1, 2, 3, 4, 5, 6};
```
Comment (bartekxk) [inline test/gemm_split_k/test_gemm_splitk_ut_cases.inc:5]: Here and in each other places: Replace int with ck::index_t

File: test/gemm_split_k/test_gemm_splitk_util.hpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    protected:
+    using ALayout   = std::tuple_element_t<0, Tuple>;
+    using BLayout   = std::tuple_element_t<1, Tuple>;
+    using CLayout   = Row;
+    using ADataType = std::tuple_element_t<2, Tuple>;
+    using BDataType = std::tuple_element_t<3, Tuple>;
+    using CDataType = std::tuple_element_t<4, Tuple>;
+
+    public:
+    bool verify_     = true;
```
Comment (bartekxk) [inline test/gemm_split_k/test_gemm_splitk_util.hpp]: Here and bellow: constexpr

File: test/grouped_gemm/test_grouped_gemm_interface.cpp [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+                                                         BBlockTransferSrcScalarPerVector,
+                                                         CDEBlockTransferScalarPerVector_NPerBlock>;
+
+    using DefaultGGemmInstance = GGemmInstance<GemmDefault, 32, 8, 4, 8, 8>;
+};
+
+TEST_F(TestGGemmSplitKInterface_MKNKMN, TileSize)
+{
+    std::vector<int> Ms{128, 256, 188, 512};
+    int N = 256;
```
Comment (bartekxk) [inline test/grouped_gemm/test_grouped_gemm_interface.cpp]: Here and bellow: constexpr

File: test/grouped_gemm/test_grouped_gemm_ut_cases.inc [hunk 4]
```diff
@@ -0,0 +1,180 @@
+#pragma once
+
+TEST_P(RRR_F16_F16_F16, TinyCases)
+{
+    const std::vector<int> Ms{0, 1};
```
Comment (bartekxk) [inline test/grouped_gemm/test_grouped_gemm_ut_cases.inc:5]: Here and bellow: constexpr

File: test/grouped_gemm/test_grouped_gemm_util.hpp [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+    protected:
+    using ALayout   = std::tuple_element_t<0, Tuple>;
+    using BLayout   = std::tuple_element_t<1, Tuple>;
+    using ELayout   = std::tuple_element_t<2, Tuple>;
+    using ADataType = std::tuple_element_t<3, Tuple>;
+    using BDataType = std::tuple_element_t<4, Tuple>;
+    using EDataType = std::tuple_element_t<5, Tuple>;
+
+    public:
+    bool verify_     = true;
```
Comment (bartekxk) [inline test/grouped_gemm/test_grouped_gemm_util.hpp]: Here and bellow: constexpr

File: test/gemm_split_k/test_gemm_splitk_util.hpp [hunk 6]
```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2023, Advanced Micro Devices, Inc. All rights reserved.
```
Comment (bartekxk) [inline test/gemm_split_k/test_gemm_splitk_util.hpp:2]: This file and each new file: Shouldn't the licenses be for 2023? As far is it new file.

Comment (aosewski) [reply test/gemm_split_k/test_gemm_splitk_util.hpp:2]: Hmm this is a good point. I think we're (repo-wide) not clear and consistent in this. We should discus the rule for it.

**Review by bartekxk — DISMISSED** [review]

**Review by zjing14 — DISMISSED** [review]

**Comment by zjing14** [comment]
@aosewski Could you resolve the conflicts?

**Comment by aosewski** [comment]
> @aosewski Could you resolve the conflicts?

@zjing14  Done

**Review by zjing14 — APPROVED** [review]

**Pull request merged**